modules/apigee_edge_apidocs/tests/src/Kernel/ApidocEntityRevisionsAccessTest.php

```diff
@@ -0,0 +1,200 @@
+<?php
+
+/**
+ * Copyright 2019 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+namespace Drupal\Tests\apigee_edge_apidocs\Kernel;
+
+use Drupal\apigee_edge_apidocs\Entity\ApiDoc;
+use Drupal\KernelTests\KernelTestBase;
+use Drupal\Tests\user\Traits\UserCreationTrait;
+
+/**
+ * Tests the ApiDoc entity access permissions.
+ *
+ * @group apigee_edge
+ * @group apigee_edge_apidocs
+ */
+class ApidocEntityRevisionsAccessTest extends KernelTestBase {
+
+  use UserCreationTrait;
+
+  /**
+   * A published API Doc.
+   *
+   * @var \Drupal\apigee_edge_apidocs\Entity\ApiDoc
+   */
+  protected $apidoc;
+
+  /**
+   * An Api Doc revision id.
+   *
+   * @var int
+   */
+  protected $apidocV1Id;
+
+  /**
+   * An Api Doc revision id.
+   *
+   * @var int
+   */
+  protected $apidocV2Id;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The entity type storage instance.
+   *
+   * @var \Drupal\Core\Entity\EntityStorageInterface
+   */
+  protected $entityTypeStorage;
+
+  protected static $modules = [
+    'system',
+    'user',
+    'text',
+    'file',
+    'apigee_edge',
+    'key',
+    'apigee_edge_apidocs',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    $this->installEntitySchema('user');
+    $this->installEntitySchema('apidoc');
+    $this->installSchema('system', ['sequences']);
+
+    $this->entityTypeManager = $this->container->get('entity_type.manager');
+    $this->entityTypeStorage = $this->entityTypeManager->getStorage('apidoc');
+
+    // Create a published apidoc.
+    $apidoc = ApiDoc::create([
```

Comment:
Avoid calling methods directly on the ApiDoc class/object, access methods via the apidoc storage. `$this->entityTypeStorage->create()`

Reply:
Fixed!